### DISCUSSION TITLE
Termux for Android 5 and 6

### DISCUSSION THREAD

**Discussion by ViswaPrabha** [termux/termux-app#2920] [General]
I have an android 6.0 phone. I would like to install termux in that phone, It is kinda super important for me to have that phone have a termux app. How can I install it?

**Comment by ViswaPrabha** [comment]
I am even willing to use older versions if you can guide on how to install that.

**Comment by agnostic-apollo** [comment]
https://github.com/termux/termux-app/wiki/Termux-on-android-5-or-6

**Reply by ViswaPrabha** [reply]
So will I be able to install it in my mobile ? If yes how ? 
Pls guide me. I'm a beginner enthusiast in tech field.

**Reply by agnostic-apollo** [reply]
Download `*apt-android-5-github-debug_universal` build from https://github.com/termux/termux-app/actions/runs/2684144105

**Reply by iTzVoko** [reply]
When I run `uname -m` command, it's tell me that I'm using armv8l.
Should I download `termux-app_v0.118.0+2113407-apt-android-5-github-debug_arm64-v8a` or `termux-app_v0.118.0+2113407-apt-android-5-github-debug_universal` ?

**Reply by ViswaPrabha** [reply]
I have no idea. Can some maintainers guide us? It will be really helpful

> When I run `uname -m` command, it's tell me that I'm using armv8l. Should I download `termux-app_v0.118.0+2113407-apt-android-5-github-debug_arm64-v8a` or `termux-app_v0.118.0+2113407-apt-android-5-github-debug_universal` ?

**Reply by agnostic-apollo** [reply]
Universal if you don't know. Otherwise `dpkg --print-architecture` should tell you.

**Reply by SmallJoker** [reply]
Thank you very much. I initially noticed that `pkg install` broke, then I noticed that I was stuck on v0.83,. In order to install a newer version, the old one had to be removed.

Specs: Android 6.0, armv7l
Build used: termux-app_v0.118.0+2113407-apt-android-5-github-debug_universal
Apps tested so far: ssh (openssl), wget (wget), links (links)

**Reply by iTzVoko** [reply]
That's good, I installed [termux-armeabi-v7a.apk](https://github.com/termux/termux-app/suites/7389433811/artifacts/300738910) because I'm using armv7 and it's smaller than universal. So download it if you want :)

**Comment by ChiefMikeK** [comment]
ARMv8-l(limited instruction set) lacks **some** 64bit [ABI - Wikipedia](https://en.wikipedia.org/wiki/Application_binary_interface ) logic/instructions vs ARMv8-A(all) is fully 64bit.

**Reply by ChiefMikeK** [reply]
Repeat:
## Download 
## *apt-android-5-github-debug_universal  
## build from   
https://github.com/termux/termux-app/actions/runs/2684144105

**Reply by iTzVoko** [reply]
Thanks!

**Reply by iTzVoko** [reply]
> ARMv8-l(limited instruction set) lacks **some** 64bit [ABI - Wikipedia](https://en.wikipedia.org/wiki/Application_binary_interface) logic/instructions vs ARMv8-A(all) is fully 64bit.

What did you mean by "limited instruction set" and "it lacks some 64bit" exactly? What can I do with it? What does it differ from ARMv7?

**Reply by sylirre** [reply]
ARMv8-l on Android typically is 32 bit operating system on a device that fully capable of 64 bit support. Manufacturers have different reasons for that, for example if design assumes the small system partition. For now all 64 bit devices required to keep set of 32 bit libraries as well for backwards compatibility, of course that increases size of /system partition.

Never seen special versions of 64 bit SoCs for Android devices which have limited instruction set.

**Reply by sylirre** [reply]
Also: l stands rather for "little endian". There are other suffixes can be present sometimes.

How LLVM treats these suffixes: https://reviews.llvm.org/D69904

Here is a StackOverflow post about this: https://stackoverflow.com/a/73696522

So basically whether one indeed could have limited execution capabilities, this doesn't mean 64bit instructions are taken away from the CPU.

**Comment by juergen874** [comment]
The apt-android-5 artifact from github CI builds have expired, can someone reactivate please ?

**Reply by agnostic-apollo** [reply]
https://github.com/termux/termux-app/actions/workflows/debug_build.yml?query=branch%3Amaster+

**Reply by tomcass240** [reply]
where do you download the apk from that link?

**Reply by sylirre** [reply]
Click on a build. Click on summary tab. Scroll down to apk list.

![Screenshot_Laptop_20240206-094029](https://github.com/termux/termux-app/assets/107305601/74af5762-8b91-420b-aa5c-9b0702756b13)

![Screenshot_Laptop_20240206-094054](https://github.com/termux/termux-app/assets/107305601/51d3e584-48e4-42a7-b431-afcda15b6758)

**Comment by al-one** [comment]
The Termux APP built by CI cannot use plugins such as Termux:Boot, because it uses a different signature from f-droid.

```
$ adb install -r Downloads/apk/com.termux.boot_7.apk
Performing Push Install
Downloads/apk/com.termux.boot_7.apk: 1 file pushed. 0.0 MB/s (9723 bytes in 0.531s)
	pkg: /data/local/tmp/com.termux.boot_7.apk
Failure [INSTALL_FAILED_SHARED_USER_INCOMPATIBLE]
```

**Reply by sylirre** [reply]
That's expected. Please check the README file of the Termux app repository.

**Reply by al-one** [reply]
Thanks for the quick reply, I found the plugins signed by Github from [here](https://archive.org/download/termux-repositories-legacy) and they work.
